### DISCUSSION TITLE
Is there a way parse a date and duration at the same time?

### DISCUSSION THREAD

**Discussion by eden-lane** [nlp-compromise/nlp_compromise#1119] [Q&A] [answered]
As I see, methods are not chainable, so I can't do something like this:
```js
compromise('tomorrow at 12:30 for 30 minutes').dates().durations().get()
```

Is there a way to do it right?

**Comment by spencermountain** [comment, marked as answer]
hey eden lane, sure it's possible to do both, depending on what you're looking for.
There's nothing wrong with calling both methods on the document:
```js
let doc = nlp('tomorrow at 12:30 for 30 minutes')
let res1 = doc.durations().json()
let res2 = doc.dates().json()

//or alternatively, if you wanted to drill-down
let m =  doc.dates()
let res1 = m.json()
let res2 = m.durations().json()
```
let me know if that helps, cheers

**Reply by eden-lane** [reply]
Aha, I thought so, but it gives me a wrong end date:
```js
{
  start: '2024-06-21T12:30:00.000+02:00',
  end: '2024-06-21T23:59:59.999+02:00',
}
```

I hoped it will get the end date based on parsed duration somehow (so it would be `2024-06-21T13:00:00.000+02:00`). 
If that's not possible, I guess I have to alter the end date myself. Thanks for the answer!
